web/packages/teleport/src/Bots/Add/GitHubActionsK8s/KubernetesLabelsSelect.tsx

```diff
@@ -0,0 +1,571 @@
+          {selected.map(l => (
+            <Label key={l.name} label={formatLabel(l)} style="secondary" />
+          ))}
+
+          {selected.length === 0 && <EmptyText>No labels selected.</EmptyText>}
+
+          <ButtonIcon
+            onClick={() => {
+              setShowPicker(true);
+            }}
+            aria-label="edit"
+          >
+            <Pencil size="medium" />
+          </ButtonIcon>
+        </LabelsContainer>
+      </LabelInput>
+
+      <HelperTextLine
+        hasError={!valid}
+        helperTextId={'labels-select-helper-text'}
+        errorMessage={message}
+      />
+
+      {showPicker && (
+        <Picker
+          selected={selected}
+          onChange={handleChanged}
+          onCancel={handleCancel}
+        />
+      )}
+    </Box>
+  );
+}
+
+const LabelsContainer = styled(Flex)`
+  flex-wrap: wrap;
+  overflow: hidden;
+  gap: ${props => props.theme.space[2]}px;
+  align-items: center;
+`;
+
+const LabelText = styled(Text).attrs({
+  typography: 'body2',
+})`
+  white-space: nowrap;
+`;
+
+function Picker(props: {
+  selected: KubernetesLabel[];
+  onChange: (labels: KubernetesLabel[]) => void;
+  onCancel: () => void;
+}) {
+  const { selected: initial, onChange, onCancel } = props;
+
+  const [selected, setSelected] = useState(initial);
+  const [manualInput, setManualInput] = useState('');
+
+  const ctx = useTeleport();
+  const flags = ctx.getFeatureFlags();
+  const hasListPermission = flags.kubernetes;
+
+  const {
+    data,
+    isLoading,
+    isFetchingNextPage,
+    error,
+    hasNextPage,
+    fetchNextPage,
+  } = useInfiniteQuery({
+    enabled: hasListPermission,
+    queryKey: ['list', 'unified_resources', cfg.proxyCluster, ['kube_cluster']],
+    queryFn: ({ pageParam, signal }) =>
+      ctx.resourceService.fetchUnifiedResources(
+        cfg.proxyCluster,
+        {
+          kinds: ['kube_cluster'],
+          startKey: pageParam,
+          limit: 20,
+        },
+        signal
+      ),
+    initialPageParam: '',
+    getNextPageParam: data => data?.startKey || undefined,
+    placeholderData: keepPreviousData,
+    staleTime: 30_000, // Cached pages are valid for 30 seconds
+  });
+
+  const handleAdd = (name: string, value: string) => {
+    setSelected(prev => {
+      const index = prev.findIndex(l => l.name === name);
+      if (index == -1) {
+        return [...prev, { name, values: [value] }];
+      } else {
+        return prev.map(l => {
+          if (l.name === name) {
+            const vs = new Set(l.values);
+            vs.add(value);
+            return {
+              ...l,
+              values: [...vs],
+            };
+          }
+          return l;
+        });
+      }
+    });
+  };
+
+  const handleManualAdd = (validator: Validator) => {
+    if (!validator.validate()) {
+      return;
+    }
+
+    const [n, v] = manualInput.split(':');
+    const name = n.trim();
+    const value = v.trim();
```

Comment:
we probably need to decide on a common separator for the label

the way i've been splitting labels is by separator `: ` (colon and blank space) and keep the key, and join rest of the result back together as one string as its `value` e.g.:

if user input `env: hello: world`, then key is `env` and value is `hello: world`

i did it this way b/c i think label `key` does not allow a space in it (i think i checked it by trying to add a label for a static database/app defined in my teleport yaml, and cluster wouldn't start b/c it said i had a space) but you can have spaces and colons in the label `value`

wdyt?